integration-cli/docker_cli_daemon_test.go

```diff
@@ -2703,7 +2703,7 @@ func (s *DockerDaemonSuite) TestExecWithUserAfterLiveRestore(c *testing.T) {
 	testRequires(c, DaemonIsLinux)
 	s.d.StartWithBusybox(c, "--live-restore")
 
-	out, err := s.d.Cmd("run", "-d", "--name=top", "busybox", "sh", "-c", "addgroup -S test && adduser -S -G test test -D -s /bin/sh && touch /adduser_end && top")
+	out, err := s.d.Cmd("run", "--init", "-d", "--name=top", "busybox", "sh", "-c", "addgroup -S test && adduser -S -G test test -D -s /bin/sh && touch /adduser_end && exec top")
```

Comment:
could you check if using `--init` correctly returns a non-zero status if this fails? I recall trying to use `--init` for all tests, but then discovered that doing so always returns a `0` exit code (because tini doesn't proxy the child processes exit code), causing some tests to fail https://github.com/moby/moby/pull/38571

Reply:
We don't really care about this exit code, it will fail later if there is a problem. But yeah can test.

Note from the tini readme:

> Remapping exit codes
> 
> Tini will reuse the child's exit code when exiting, but occasionally, this may not be exactly what you want (e.g. if your child exits with 143 after receiving SIGTERM). Notably, this can be an issue with Java apps.
>
> In this case, you can use the -e flag to remap an arbitrary exit code to 0. You can pass the flag multiple times if needed.

Reply:
>  docker run --rm --init busybox sh -c 'exit 1'; echo $?
1

Reply:
👍 thanks! Perhaps it was one test in particular that didn't like running with `--init` 🤔